typescript/testsuite/network_impl/test_network.ts

```diff
@@ -0,0 +1,350 @@
+        }
+        if (!addServiceResult.isOk()) {
+            return err(addServiceResult.error);
+        }
+        const apiServiceContext: ServiceContext = addServiceResult.value[0];
+        const hostPortBindings: Map<string, PortBinding> = addServiceResult.value[1];
+    
+        const apiClient: APIClient = new APIClient(apiServiceContext.getIPAddress(), API_SERVICE_PORT);
+    
+        let apiClientWaitForHealthyResult: Result<null, Error>;
+        try {
+            apiClientWaitForHealthyResult = await apiClient.waitForHealthy(WAIT_FOR_STARTUP_MAX_NUM_POLLS, WAIT_FOR_STARTUP_DELAY_MILLISECONDS);
+        } catch(exception: any) {
+            // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+            // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+            if (exception && exception.stack && exception.message) {
+                return err(exception as Error);
+            }
+            return err(new Error("Calling waitForHealthy method on APIClient class threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+        if (!apiClientWaitForHealthyResult.isOk()) {
+            return err(apiClientWaitForHealthyResult.error);
+        }
+    
+        log.info("Added API service with host port bindings:", hostPortBindings)
+        return ok(apiClient);
+    }
+
+    static getDataStoreContainerCreationConfig(): ContainerCreationConfig {
+        const usedPortsSet: Set<string> = new Set();
+        const containerCreationConfig: ContainerCreationConfig = new ContainerCreationConfigBuilder(
+            DATASTORE_IMAGE,
+        ).withUsedPorts(
+            usedPortsSet.add(DATASTORE_PORT+"/tcp")
+        ).build()
+        return containerCreationConfig;
+    }
+
+    static getDataStoreRunConfigFunc(): (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> {
+        const runConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = 
+        (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => {
+            return ok(new ContainerRunConfigBuilder().build());
+        }
+        return runConfigFunc;
+    }
+
+    static async getApiServiceConfigInitializingFunc(datastoreClientResult: Result<DatastoreClient, Error>): Promise<(fp: number) => Promise<Result<null, Error>>> { //Making simplification that file descriptor is just number
+        const configInitializingFunc: (fp: number) => Promise<Result<null, Error>> = async (fp: number) => { //TOOD (Ali) - might require changes in ConfigRunFactory in kurt-client
+            if (!datastoreClientResult.isOk()) {
```

Comment:
Like from above - we should never pass a `Result<>` as an arg into a function; we should check its error state and unpack it immediatley after we get it